packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -150,7 +190,26 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
   /// See also:
   ///
   ///  * [AnnotatedRegionLayer], for placing values in the layer tree.
-  Iterable<S> findAll<S>(Offset regionOffset);
+  Iterable<S> findAll<S>(Offset regionOffset) {
+    final LayerHitTestResult<S> result = LayerHitTestResult<S>();
+    hitTest<S>(result, regionOffset: regionOffset);
+    return result.path.map((LayerHitTestEntry<S> entry) {
+      return entry.value;
+    });
+  }
+
+  /// Determines the list of annotations located at the given position.
+  ///
+  /// It adds any annotations of the given type `T` that contain the point to
+  /// the given layer hit test result.
+  ///
+  /// It returns true if this layer or one of its descendants absorbs the hit
+  /// (preventing non-ancestral layers below this one from being hit).
+  /// Returns false if the hit can continue to other non-ancestral objects
+  /// behind this one.
+  bool hitTest<S>(LayerHitTestResult<S> result, { @required Offset regionOffset }) {
```

Comment:
would be nice for this to be abstract, so that anyone creating a new layer is encouraged to implement it